app/kubemci/pkg/validations/validations.go

```diff
@@ -89,3 +91,69 @@ func nodePortSameInAllClusters(backend v1beta1.IngressBackend, namespace string,
 	}
 	return nil
 }
+
+func ServerVersionsNewEnough(clients map[string]kubeclient.Interface) (bool, error) {
+	for key := range clients {
+		glog.Infof("Checking client %s", key)
+		discoveryClient := clients[key].Discovery()
+		if discoveryClient == nil {
+			return false, fmt.Errorf("no discovery client in client: %+v", clients[key])
+		}
+		ver, err := discoveryClient.ServerVersion()
+		if err != nil {
+			return false, fmt.Errorf("could not get discovery client to lookup server version: %s", err)
+		}
+		glog.Infof("ServerVersion: %+v", ver)
+		newEnough, err := serverVersionNewEnough(ver.GitVersion)
+		if err != nil {
+			return false, fmt.Errorf("error checking server version: %s", err)
+		}
+		if !newEnough {
+			fmt.Println("Cluster", key, "is not running a supported kubernetes version. Need at least 1.8.1.")
+			return false, nil
+		}
+
+	}
+	return true, nil
+}
+
+func serverVersionNewEnough(version string) (bool, error) {
```

Comment:
Add a unit test for this.
Recommend splitting this into
parseServerVersion() and validateServerVersion() and then you can test both independently.

Reply:
oops. I wrote a test in validations_test.go, but somehow it got lost before uploading this PR.